docs-content/getting-started/fullstack-frameworks/next-js/1_overview.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Next.js Overview
+slug: overview
+heading: Next.js Overview
+createdAt: 2023-05-10T00:00:00.000Z
+updatedAt: 2023-05-10T00:00:00.000Z
+---
+
+<EmbeddedVideo 
+  src="https://www.youtube.com/embed/Dyoba16wE-o"
+  title="Youtube Video Player"
+  allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+/>
+
+## Overview
+
+Our Next.js SDK gives you access to frontend session replays and server-side monitoring,
+all-in-one. 
+
+1. On the frontend, the `<HighlightInit/>` component sets up client-side session replays.
+2. On the backend, the `PageRouterHighlight` wrapper exported from `@highlight-run/next/server` captures server-side errors and logs from Page Router API endpoints.
+3. On the backend, the `AppRouterHighlight` wrapper exported from `@highlight-run/next/app-router` captures errors and logs from App Router API endpoints.
+3. The `EdgeHighlight` wrapper exported from `@highlight-run/next/server` captures server-side errors and logs from both Page and App Router endpoints using Vercel's Edge runtime.
+4. Use `pages/_error.tsx` and `app/error.tsx` to forward Page Router and App Router SSR errors from the client to Highlight.
+5. The `withHighlightConfig` configuration wrapper automatically proxies Highlight data to bypass ad-blockers and uploads source maps so your frontend errors include stack traces to your source code.
+
+### How Highlight captures Next.js errors
+
+|              | Page Router           | App Router           |
+|--------------|-----------------------|----------------------|
+| API Errors   | `PageRouterHighlight` | `AppRouterHighlight` |
+| SSR Errors   | `pages/_error.tsx`    | `app/error.tsx`      |
+| Client       | `<HighlightInit />`   | `<HighlightInit />`  |
+| Edge runtime | `EdgeHighlight`       | `EdgeHighlight`      |
+
+### How Highlight captures Next.js logs
+
+`<HighlightInit />` captures front-end logs.
+
+`PageRouterHighlight` and `AppRouterHighlight` capture server-side logs in traditional server runtimes. These wrappers typically fail in serverless runtimes (including Vercel), because we cannot guarantee that the serverless process will stay alive long enough to send all log data to Highlight.
+
+Configure logging for your serverless cloud provider using one of our [cloud provider logging guides](https://www.highlight.io/docs/getting-started/backend-logging/hosting/overview), including [Vercel Log Drain for Highlight](https://vercel.com/integrations/highlight).
+
+## Next Steps
+
+- [Environment (optional)](./2_environment.md)
```

Comment:
i still think this env setup is a bit distracting, we should just link straight to the client setup and environment can be a follow up after u have things working

Reply:
I agree, but I'm torn. The nice thing about using `constants.ts` is that it maintains consistency with our `e2e/nextjs` files, so I can copy/paste right out of the real files. I also like that it shows less-experienced Next.js the best way to inject environment variables, because most will just copy/paste their strings throughout their codebases.

I could keep the code samples the way they are and just hide the environment stuff??

What feels best to you?

Reply:
I could also hide the environment example at the end under "configuration".

Reply:
I'll leave this for now. It's easy to delete later if it causes confusion. Theo may have some opinions.